Add PSGO

HoeenCoder opened the pull request:
Credit to @Arrays-Arr as well

(Please review)

Lycanium-Z commented on wavelength-plugins/psgo.js:
```diff
@@ -0,0 +1,545 @@
+		},
+		ladderhelp: ['/psgo ladder - show the PSGO card point ladder.'],
+
+		nuke: 'reset',
```
>nuke

HoeenCoder replied:
Something wrong with this as an alias? or just commenting on it

DeathlyPlays replied:
Yeah idk what he means

Lycanium-Z replied:
Just commenting

DeathlyPlays commented on wavelength-plugins/factions.js:
```diff
@@ -49,47 +49,6 @@ function getFactionRank(user) {
 
-// SHAMLESSLY RIPPED FROM ECONOMY.JS!
-
-function rankLadder(title, type, array, prop, group) {
```
Don't forget exp uses it

DeathlyPlays replied:
And ontime I've been meaning to make that a global tbh

HoeenCoder replied:
done

DeathlyPlays commented on wavelength-plugins/psgo.js:
```diff
@@ -0,0 +1,545 @@
+
+		/*
+			TODO: Custom cards
+			Will be stored in a seperate JSON file
```
Separate*

DeathlyPlays commented on wavelength-plugins/psgo.js:
```diff
@@ -0,0 +1,545 @@
+				r.addRaw(`<div class="broadcast-red"><strong>The PSGO database was reset</strong><br/>You no longer have any cards or packs.`).update();
+			});
+		},
+		resethelp: ['/psgo reset - Wipe the PSGO database (Take ALL users cards AND packs). Requires: ~'],
```
Nitpicky but user's

HoeenCoder replied:
I don't think thats correct in this case actually, since were not refering to one user.

DeathlyPlays commented on wavelength-plugins/psgo.js:
```diff
@@ -0,0 +1,545 @@
+		'/psgo showcase (user) - Show all of the selected users cards.',
+		'/psgo transfercard [user], [card ID] - Transfer one of your cards to another user.',
+		'/psgo cardsearch - sends a display to search for a list of cards.',
+		'/psgo give user, card - gives the user specified card. Requires &, ~',
```
For consistency brackets around user and card

DeathlyPlays commented on wavelength-plugins/psgo.js:
```diff
@@ -0,0 +1,545 @@
+		'/psgo takeall [user] - takes all cards from the specified user. Requires: &, ~',
+		'/psgo shop buy - buys a pack from the pack shop. Costs 5 ' + global.currencyPlural + '.',
+		'/psgo shop display - shops pack shop.<br />',
+		'/psgo packs give user, pack name - gives a user a  pack. Requires &, ~',
```
Consistency with user and pack name in brackets

HoeenCoder commented:
Searching cards already has an alias

giving/taking dosent really need one as its not often they will be used (&+)

I'll add one for opening

HoeenCoder commented:
- Fixed above issues
- Added shortcuts for opening packs, and checking your packs
- Added a function to turn a string into a pack name (since pack names need to be written exactly right to work). Basically it takes something like this ``bLAckwhITE`` and turns it into the correct pack name: ``Black & White``
- cardsearch now supports the two card types (actual type like water, and type like EX, Basic, Stage 1, ect)
- cardseach is now limited to 500 cards shown per search, this is so users client aren't lagged by 1000+ cards when they do a broad search like any common card.

DeathlyPlays commented:
Good idea because cards is typically laggy

DeathlyPlays commented:
I would suggest a command that adds cards to the json file @HoeenCoder

DeathlyPlays commented:
And make a card leader sorta thing like vouchers and allow C733937123 to be the 1st Card Leader imo @HoeenCoder

HoeenCoder commented:
@DeathlyPlays thats planned, but its going to be a seperate JSON file. I'll do it soon-ish (i wanted this up for review first)

DeathlyPlays commented:
True; later expansions are nice to keep hype going
